modules/fundamental/src/vulnerability/service/test.rs

```diff
@@ -427,3 +427,48 @@ async fn vulnerability_queries(ctx: &TrustifyContext) -> Result<(), anyhow::Erro
 
     Ok(())
 }
+
+#[test_context(TrustifyContext)]
+#[test(actix_web::test)]
+async fn analyze_purls(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
+    let service = VulnerabilityService::new();
+
+    ctx.ingest_documents(["osv/RUSTSEC-2021-0079.json", "osv/GHSA-2ccf-ffrj-m4qw.json"])
+        .await?;
+
+    let mut result = service.analyze_purls(vec![], &ctx.db).await?;
+
+    assert!(result.is_empty());
+
+    let expected = [
+        "pkg:npm/@fastify/passport@1.0.0", // ECOSYSTEM:afected
```

Comment:
This is an invalid purl. It should've been percent-encoded [here](https://github.com/trustification/trustify/blob/5e1075324e8821c8a636725c7da004127882cd2f/common/src/purl.rs#L289), or better yet, we should create a `new` fn for our [Purl struct](https://github.com/trustification/trustify/blob/5e1075324e8821c8a636725c7da004127882cd2f/common/src/purl.rs#L33-L39) and hide all its fields.

Reply:
updated.